### PR TITLE
feat(conn init): make origin optional

### PR DESCRIPTION
**Pull request by itayd**
makes it easier to start stuff from the dashboard and debug.

### CODE REVIEW

File: internal/backend/oauth/oauth.go [hunk 1]
```diff
@@ -413,10 +413,6 @@ func (o *oauth) StartFlow(ctx context.Context, intg string, cid sdktypes.Connect
 		return "", errors.New("invalid connection ID")
 	}
 
-	if origin == "" {
```
Comment (daabr) [reply internal/backend/oauth/oauth.go:416]: To answer @efiShtain - there's no security aspect

File: internal/backend/oauth/webhook.go [hunk 2]
```diff
@@ -85,7 +85,7 @@ func (s *svc) exchange(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Ensure the state parameter is well-formed.
-	sub := regexp.MustCompile(`^(.+)_([a-z]+)$`).FindStringSubmatch(state)
```
Comment (daabr) [inline internal/backend/oauth/webhook.go:88]: The change from `+` to `*` is incorrect because the `state` param is supposed to be an AK ID.

**Review by daabr — CHANGES_REQUESTED** [review]
The `origin` param is supposed to be a required parameter - the specified value affects the way we end connection flows depending on whether they were initiated by the web UI / CLI / dashboard / etc. So currently there is no "default origin", and I don't think it's appropriate to have one, but it there was it wouldn be the web UI and not the dashboard.

File: internal/backend/oauth/oauth.go (hunk 1 above)
Comment (efiShtain) [inline internal/backend/oauth/oauth.go:416]: do we use it for any security aspect ?

**Pull request closed without merging**